### ISSUE TITLE
[BUG] DataFrame mutates Column arguments

### ISSUE THREAD

**Issue by trxcllnt** [rapidsai/cudf#2181]
**Describe the bug**
DataFrame now adds a `_name` to each column that comes in. It shouldn't mutate the instances passed in, but replace the name with the desired name via `col.replace({name})`.

**Steps/Code to reproduce bug**
```python
col = as_column(['a', 'b', 'c'])
DataFrame({'wat': col})
print(col._name)
# wat
```

**Expected behavior**
The column is not mutated.

**Comment by shwina** [comment]
In general, having two distinct `name` attributes leads to confusion:

```
In [1]: import cudf                                 

In [2]: a = cudf.Series([1, 2, 3], name='a')        

In [3]: b = cudf.DataFrame({'b': a})                

In [4]: print(b)                                    
   b
0  1
1  2
2  3

In [5]: print(a)                                    
0    1
1    2
2    3
Name: a, dtype: int64

In [6]: a._column.name                              
Out[6]: 'b'

In [7]: a.name                                      
Out[7]: 'a'
```

**Comment by kkraus14** [comment]
Fixed by #2545
